SDK v2 migration guide.md

```diff
@@ -16,182 +16,189 @@ are outlines of the notable breaking changes as well as a mapping from version 1
-| `Message.EnqueuedTimeUtc` | `TelemetryMessage.EnqueuedtimeUtc`, `IncomingMessage.EnqueuedTimeUtc` |
-| `Message.ExpiryTimeUtc` | `TelemetryMessage.ExpiresOnUtc`, `IncomingMessage.ExpiresOnUtc` |
-| `MethodRequest` | `DirectMethodRequest` |
-| `MethodResponse` | `DirectMethodResponse` |
-| `DeviceAuthenticationWithRegistrySymmetricKey`, `DeviceAuthenticationWithSharedAccessPolicyKey`, `ModuleAuthenticationWithRegistrySymmetricKey` | `ClientAuthenticationWithSharedAccessKeyRefresh` |
-| `DeviceAuthenticationWithToken`, `ModuleAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` |
-| `DeviceAuthenticationWithTokenRefresh`, `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithTokenRefresh` |
-| `DeviceAuthenticationWithX509Certificate` | `ClientAuthenticationWithX509Certificate` |
-
-#### Other notable breaking changes
+#### Notable breaking changes
 
 - The transport default has changed from AMQP (TCP with web socket fail over) to MQTT TCP.
   - To override the transport default, create an instance of `IotHubClientOptions` and pass an instance of the transport settings you wish to use (i.e., `IotHubClientMqttSettings`, `IotHubClientAmqpSettings`).
   - TCP will be the default. For web socket, pass `IotHubClientTransportProtocol.WebSocket` to the transport settings constructors.
 - HTTP has been removed as a transport option.
-  - It had very limited support across the device options and some APIs behaved differently.
+  > It had very limited support across the device options and some APIs behaved differently.
+  > We feel this caused more harm than good and as such it was best to remove.
 - Some options that were previously set in the `DeviceClient` constructor are now in the optional `IotHubClientOptions` parameter.
 - The connection status callback parameters have changed. Instead of two parameters, a class with several properties is provided.
-  - Two properties are the same as before, but with some renames (underscores removed) and obsolete members removed.
+  - Two properties enum values are mostly the same as before, with some renames (underscores removed) and obsolete members removed.
   - A new property has been added with a recommended action, which a device developer may observe or ignore.
 - The file upload method has been split into the three individual steps that this method used to take. See [this file upload sample](./iothub/device/samples/getting%20started/FileUploadSample/) for an example of how to do file upload using these discrete steps.
 - Cloud-to-device messages can be received by calling `SetMessageHandlerAsync` and providing a callback. Users no longer need to poll for messages with `ReceiveAsync`.
-- Several callback handler set methods and definitions have changed, losing the `userContext` parameter.
+- Several callback handler set methods and definitions have changed, losing the `userContext` parameter which was deemed unnecessary and a vestige of the C device client.
```

Comment:
Ah, I didn't know it was a C client pattern.

Reply:
Pretty sure. I've seen that pattern for years in C, back before C# existed.